src/shared/analytics/analytics.background.ts

```diff
@@ -40,21 +40,36 @@ function trackAppEvents({ account }: { account: Account }) {
 
   const getProvider = (address: string) =>
     getProviderForMetabase(queryWalletProvider(account, address));
+  const getCurrentAddress = () =>
+    account.getCurrentWallet().readCurrentAddress();
 
-  const createParams: typeof createBaseParams = (params) => {
-    const getUserId = () => account.getUser()?.id;
-    return createBaseParams({ ...params, userId: getUserId() });
+  const createParams: typeof createBaseParams = (eventParams) => {
+    const params: typeof eventParams & {
+      wallet_address?: string | null;
+      wallet_provider?: string | null;
+    } = eventParams;
+
+    if (!params.wallet_address) {
+      params.wallet_address = getCurrentAddress();
```

Comment:
We shouldn't send null here. I see that types for getCurrentAddress allow string | null

Reply:
We have been sending it already
I'll confirm with Anya

Reply:
For mixpanel we're getting rid of all nullish params before sending: https://github.com/zeriontech/zerion-wallet-extension/blob/8ec927fa7ecf45921219bdcad41651bf09bacf8c/src/shared/analytics/mixpanel.ts#L37-L47